lib/Controller/ChattyLLMController.php

```diff
@@ -448,9 +451,29 @@ private function getStichedMessages(int $sessionId): string {
 	 *
 	 * @param string $content
 	 * @return string
+	 * @throws \OCP\TaskProcessing\Exception\Exception
+	 * @throws NotFoundException
+	 * @throws PreConditionNotMetException
+	 * @throws UnauthorizedException
+	 * @throws ValidationException
 	 */
 	private function queryLLM(string $content): string {
-		$task = new TextProcessingTask(FreePromptTaskType::class, $content, Application::APP_ID, $this->userId);
-		return trim($this->textProcessingManager->runTask($task));
+		$task = new Task(TextToText::ID, ['input' => $content], Application::APP_ID, $this->userId, 'chatty-llm');
+		$this->taskProcessingManager->scheduleTask($task);
+		$i = 0;
+		while ($i < 60 && !in_array($task->getStatus(), [Task::STATUS_SUCCESSFUL, Task::STATUS_FAILED, Task::STATUS_CANCELLED])) {
+			sleep(1);
+			$task = $this->taskProcessingManager->getTask($task->getId());
```

Comment:
As discussed, should probably poll in the browser

Reply:
Done